### DISCUSSION TITLE
How to remediate the wide prediction interval?

### DISCUSSION THREAD

**Discussion by jpzhangvincent** [lindeloev/mcp#94] [Q&A]
I'm still learning Bayesian methods and trying to apply in forecasting problems. I'm wondering whether there's a way to remediate the wide credible interval in the forecast horizon. Any prior we can incorporate to correct for that?
<img width="579" alt="Screen Shot 2020-08-27 at 10 51 03 AM" src="https://user-images.githubusercontent.com/7220686/91477551-5c7f5600-e853-11ea-8978-8e542c0a560d.png">

It seems to me that the model fits well and converges (although not perfect). How would you recommend for debugging/improving the model?
<img width="576" alt="Screen Shot 2020-08-27 at 11 03 39 AM" src="https://user-images.githubusercontent.com/7220686/91478594-014e6300-e855-11ea-8459-1e79e1069bdd.png">

**Comment by lindeloev** [comment]
Sorry I didn't notice your post here! Here's three solutions off the top of my head:

 1. Accept it :-) It literally is just regression on the data points in the last segment, so this is how all regression models with weak priors would behave in that segment.
 2. If the width of the interval is at odds with you knowledge about the future, incorporate your knowledge by setting more informative priors for the slope in the last segment. [Read more here](https://lindeloev.github.io/mcp/articles/priors.html).
 3. You can also scroll down in [this issue](https://github.com/lindeloev/mcp/issues/78) where I suggest a method for making predictions involving future change points. With so many "rapid" change points, perhaps it's unrealistic to model one unbroken segment for such a long forecast horizon?
